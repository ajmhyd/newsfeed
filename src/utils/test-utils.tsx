import React, { ReactElement, ReactNode } from 'react'
import { render } from '@testing-library/react'

type AllTheProvidersProps = {
  children: ReactNode,
}

const AllTheProviders = ({ children }: AllTheProvidersProps) => {
  return <>{children}</>
}

const customRender = (ui: ReactElement, options?: OptionalType) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
