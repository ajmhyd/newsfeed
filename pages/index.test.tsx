import { render, screen } from 'utils/test-utils';
import Index from './index';

test('renders', () => {
  const index = render(<Index />);
  expect(index).toMatchSnapshot();
})
