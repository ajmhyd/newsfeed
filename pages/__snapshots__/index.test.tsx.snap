// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="bg-gray-800"
        >
          <div
            class="max-w-7xl mx-auto px-2 sm: px-6 lg:px-8"
          >
            <div
              class="relative flex items-center justify-center h-16"
            >
              <span
                class="text-lg text-white font-bold"
              >
                NewsFeed
              </span>
            </div>
          </div>
        </nav>
        <div
          class="container mx-auto max-w-3xl pt-6"
        >
          <div
            class="bg-white shadow overflow-hidden sm:rounded-md"
          >
            <ul
              class="divide-y divide-gray-200"
            >
              Loading...
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="bg-gray-800"
      >
        <div
          class="max-w-7xl mx-auto px-2 sm: px-6 lg:px-8"
        >
          <div
            class="relative flex items-center justify-center h-16"
          >
            <span
              class="text-lg text-white font-bold"
            >
              NewsFeed
            </span>
          </div>
        </div>
      </nav>
      <div
        class="container mx-auto max-w-3xl pt-6"
      >
        <div
          class="bg-white shadow overflow-hidden sm:rounded-md"
        >
          <ul
            class="divide-y divide-gray-200"
          >
            Loading...
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
